components/bitcoin-topics/markdown-ui/opcode-data-visualizer.tsx

```diff
@@ -0,0 +1,465 @@
+                        example: "OP_PUSHDATA1 4c "
+                    }
+                    break
+                case "op_push_data_2":
+                    data = {
+                        opcode: "OP_PUSHDATA2",
+                        description: "Data between 256-65535 bytes",
+                        details:
+                            "The OP_PUSHDATA2 opcode (0x4d) works in the same was as OP_PUSHDATA1, except it's followed by 2 bytes to indicate the number of upcoming bytes to be pushed on to the stack.  This is used when you want to push more data on the stack than you can with OP_PUSHDATA1.",
+                        example: "OP_PUSHDATA2 0001 "
+                    }
+                    break
+                case "op_push_data_4":
+                    data = {
+                        opcode: "OP_PUSHDATA4",
+                        description: "Data between 65536-4294967295 bytes",
+                        details:
+                            "The OP_PUSHDATA4 opcode (0x4e) works in the same was as OP_PUSHDATA2, but is followed by 4 bytes to indicate the number of upcoming bytes to be pushed on to the stack.  This is used when you want to push more data on the stack than you can with OP_PUSHDATA2.",
+                        example: "OP_PUSHDATA4 00000100 [65536 bytes]"
+                    }
+                    break
+                default:
+                    data = null // Hide panel if clicking on a different area
+            }
+
+            setInfo((prevInfo) =>
+                prevInfo?.opcode === data?.opcode ? null : data
+            )
+        }
+
+        if (svgElement) {
+            const clickableAreas = svgElement.querySelectorAll(
+                "#op_push_bytes, #op_push_data_1, #op_push_data_2, #op_push_data_4"
+            )
+            clickableAreas.forEach((area) => {
+                area.addEventListener("click", handleClick)
+            })
+        }
+
+        return () => {
+            if (svgElement) {
+                const clickableAreas = svgElement.querySelectorAll(
+                    "#op_push_bytes, #op_push_data_1, #op_push_data_2, #op_push_data_4"
+                )
+                clickableAreas.forEach((area) => {
+                    area.removeEventListener(
+                        "click",
+                        handleClick as EventListener
+                    )
+                })
+            }
+        }
+    }, [])
+
+    return (
+        <div className="flex max-w-4xl flex-col items-center p-4">
+            <p className="mb-0 text-center text-gray-600">
+                *Click on the colored rectangles to learn more about each
+                OP_PUSH data type.
+            </p>
+            <svg
+                ref={svgRef}
+                width="863"
```

Comment:
on smaller screens, the svg overflows and breaks screen layout, image below
<img width="466" alt="Screenshot 2024-08-31 at 1 31 57 AM" src="https://github.com/user-attachments/assets/9d2c0452-7139-424f-b01c-ac8226e1a5d2">

```suggestion
                width={window.innerWidth < 520 ? "120%" : "100%"}
                height={"auto"}
```
this is a rough draft to get the SVG large enough to be visible, keep aspect ratio, and not break screen layout on small screens. You can try other methods if this doesn't scale as well as you'd prefer.

Also, the `p-4` on the parent container should be removed (maximizing all the space on a small screen 🙃)